java/arcs/core/storage/CapabilitiesResolver.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.storage
+
+import androidx.annotation.VisibleForTesting
+import arcs.core.common.ArcId
+import arcs.core.data.Capabilities
+import arcs.core.util.TaggedLog
+
+/**
+ * [CapabilitiesResolver] is a factory class that creates [StorageKey]s based on the registered
+ * providers and given [Capabilities].
+ */
+class CapabilitiesResolver(
+    val options: StorageKeyOptions,
+    var creators: StorageKeyCreatorsMap = mutableMapOf()
+) {
+
+    private val log = TaggedLog { "CapabilitiesResolver" }
+
+    init {
+        if (creators.isEmpty()) {
+            CapabilitiesResolver.defaultCreators.forEach {
+                protocol, creator -> creators[protocol] = creator
+            }
+            CapabilitiesResolver.registeredCreators.forEach {
+                protocol, creator -> creators[protocol] = creator
+            }
+        }
+    }
+
+    data class StorageKeyOptions(val arcId: ArcId)
+    data class CapabilitiesCreator(
```

Comment:
Seems odd to have a data class called `*Creator`

Reply:
got rid of this data class altogether, used a Pair instead.